cdk/domino_cdk/provisioners/vpc.py

```diff
@@ -88,23 +140,51 @@ def provision_vpc(self, name: str, vpc: config.VPC):
                 subnets=ec2.SubnetSelection(subnet_type=ec2.SubnetType.PRIVATE),
             )
 
-    def provision_bastion(self, name: str, bastion: config.VPC.Bastion) -> None:
+        # TODO until https://github.com/aws/aws-cdk/issues/14194
+        for idx, subnet_id in enumerate(self.vpc.select_subnets(subnet_type=ec2.SubnetType.PUBLIC).subnet_ids):
+            name = f"DisableMapPublicIpLaunch-{idx}"
+            cr.AwsCustomResource(
+                self.scope,
+                name,
+                log_retention=logs.RetentionDays.ONE_DAY,
+                policy=cr.AwsCustomResourcePolicy.from_sdk_calls(resources=cr.AwsCustomResourcePolicy.ANY_RESOURCE),
+                on_create=cr.AwsSdkCall(
+                    action="modifySubnetAttribute",
+                    service="EC2",
+                    parameters={"MapPublicIpOnLaunch": {"Value": False}, "SubnetId": subnet_id},
+                    physical_resource_id=cr.PhysicalResourceId.of(name),
+                ),
+            )
+
+        if vpc.flow_logging:
+            if not flow_log_bucket:
+                # raise?
+                pass
+
+            if flow_log_bucket:
+                self.vpc.add_flow_log(
+                    "rejectFlowLogs",
+                    destination=ec2.FlowLogDestination.to_s3(flow_log_bucket),
+                    traffic_type=ec2.FlowLogTrafficType.REJECT,
+                )
+
+    def provision_bastion(self, name: str, bastion: config.VPC.Bastion) -> Optional[ec2.SecurityGroup]:
         if not bastion.enabled:
             return None
         if bastion.ami_id:
             machine_image = ec2.MachineImage.generic_linux(
+                # TODO no region???
                 {self.region: bastion.ami_id},
                 user_data=ec2.UserData.custom(bastion.user_data),
             )
         else:
+            # ??? is this now covered under the config? is it not auto?? why is it in here?
```

Comment:
Ostensibly, you could synth a generic cloudformation template rather than a deployment-specific one by not providing a region or account id, and I tried to keep that an option early on. Was also going to have a few pre-fab cloudformation templates. For various reasons this really didn't work out, but the bastion support was the first thing that really hard broke the possibility of doing this.

So, conceptually, if you had bastion off, you could not provide the aws account id and still proceed.

I think it's fine to move this check to config and just make it a hard requirement.